mllib/src/main/scala/org/apache/spark/mllib/tree/model/InformationGainStats.scala

```diff
@@ -76,3 +77,52 @@ private[spark] object InformationGainStats {
   val invalidInformationGainStats = new InformationGainStats(Double.MinValue, -1.0, -1.0, -1.0,
     new Predict(0.0, 0.0), new Predict(0.0, 0.0))
 }
+
+class ImpurityStats(
+    val gain: Double,
+    val impurity: Double,
+    val impurityCalculator: ImpurityCalculator,
+    val leftImpurityCalculator: ImpurityCalculator,
+    val rightImpurityCalculator: ImpurityCalculator) extends Serializable {
+
+  override def toString: String = {
+    s"gain = $gain, impurity = $impurity, left impurity = $leftImpurity, " +
+      s"right impurity = $rightImpurity"
+  }
+
+  def leftImpurity: Double = leftImpurityCalculator.calculate()
+
+  def rightImpurity: Double = rightImpurityCalculator.calculate()
+
+  override def equals(o: Any): Boolean = o match {
```

Comment:
Do you need this and hashCode?  (Are these used in data structures requiring equals/hashCode?)  If so, do you need to override equals and hashCode in the ImpurityCalculators as well?